app/scripts/components/exploration/hooks/use-analysis-data-request.ts

```diff
@@ -107,23 +114,39 @@ export function useAnalysisDataRequest({
     };
 
     const { start, end } = selectedInterval;
-
-    requestDatasetTimeseriesData({
-      maxItems: MAX_QUERY_NUM,
-      start,
-      end,
-      aoi,
-      dataset,
-      queryClient,
-      concurrencyManager: analysisConcurrencyManager,
-      onProgress: (data) => {
-        setAnalysis(data);
-      }
-    });
+    async function makeCall(){
+      const stat = await requestDatasetTimeseriesData({
+        maxItems: MAX_QUERY_NUM,
+        start,
+        end,
+        aoi,
+        dataset,
+        queryClient,
+        concurrencyManager: analysisConcurrencyManager,
+        onProgress: (data) => {
+          setAnalysis(data);
+        }
+      });
+      setAnalysisResult(stat);
+    }
+    makeCall();
```

Comment:
⛏️  : In this case, we can probably have it self invoke so we dont need to do `makeCall()` here
```
(async  () => {
    ....
})();
```